src/Forms/NumericField.php

```diff
@@ -48,6 +55,14 @@ class NumericField extends TextField
      */
     protected $scale = 0;
 
+    public function __construct($name, $title = null, $value = null, $maxLength = null, $form = null)
+    {
+        // This constructor has a default value of null for the $value param, as opposed to the parent
+        // TextField constructor which has a default value of blank string. This is done to prevent passing
+        // a blank string to the NumericFieldValidator which which will cause a validation error.
```

Comment:
> Form fields should probably be robust to string/numeric conversions

cast() method is already doing this, at least for numeric fields.  If we do decide to ditch the strong typing here, correct place to do it would be at the DBField level, not the FormField level, I think

Discussion around `null` casting on empty string can be had here https://github.com/silverstripe/silverstripe-framework/pull/11449#discussion_r1851171579

Reply:
>> Form fields should probably be robust to string/numeric conversions
>
> cast() method is already doing this

Except it isn't in this case, which is exactly my point.

Reply:
It is casting numeric strings already?  Any non-zero length string is being cast to either (int) or (float). Code for NumericField::cast() is

```php
        if (!is_string($value) || strlen($value) === 0) {
            return $value;
        }
        if ($this->getScale() === 0) {
            return (int) $value;
        }
        return (float) $value;
```

Reply:
The latest changes to the `cast()` method and how it deals with null have made this conversation unnecessary 👍